### PR TITLE
Add connect GitHub account button

### PR DESCRIPTION
**Pull request by jbranchaud**
This PR does two things:

1. It moves the Subscription Details mechanics and JSX into a separate component to declutter the `user/index` component.

2. It adds a 'Connect to GitHub' section with a button for connecting your github account to your existing egghead account.

TODO: Once your account has been 'connected' to GitHub (or if it has already been connected because that's how you originally signed up), then the button should be disabled.

CleanShot 2021-02-02 at 09 23 13@2x

### CODE REVIEW

**Comment by jbranchaud** [comment]
Taking a second look at that screenshot, the blue button is coming off too strong. I think I should make it a white/off-gray button. It's a benign, non-essential action, not exactly a CTA we are trying to draw attention to.

File: src/components/users/subscription-details.tsx [hunk 1]
```diff
@@ -0,0 +1,163 @@
+import React from 'react'
```
Comment (jbranchaud) [inline src/components/users/subscription-details.tsx:1]: Moved all this from `src/pages/user/index.tsx` into here

File: src/pages/user/index.tsx [hunk 2]
```diff
@@ -46,143 +32,35 @@ const User: React.FunctionComponent<
...
+                >
+                  Connect your GitHub account
+                </a>
+              </div>
+            </div>
           </div>
+          <SubscriptionDetails
+            stripeCustomerId={stripe_customer_id}
+            slug={slug}
+          />
```
Comment (jbranchaud) [inline src/pages/user/index.tsx:63]: Reference to new SubscriptionDetails component defined in the other file.

File: src/pages/user/index.tsx [hunk 3]
```diff
@@ -46,143 +32,35 @@ const User: React.FunctionComponent<
...
+              <p>Connect your GitHub account to log in with GitHub Oauth.</p>
+              <div>
+                <a
+                  href={`${process.env.NEXT_PUBLIC_AUTH_DOMAIN}/users/github_passthrough?client_id=${process.env.NEXT_PUBLIC_CLIENT_ID}&auth_token=${authToken}`}
+                  className="text-white bg-blue-600 border-0 py-2 px-4 focus:outline-none hover:bg-blue-700 rounded"
+                >
+                  Connect your GitHub account
+                </a>
+              </div>
+            </div>
```
Comment (jbranchaud) [inline src/pages/user/index.tsx:58]: This is the entire 'Connect to GitHub' section

**Review by joelhooks — APPROVED** [review]

**Pull request merged**